src/main/java/org/cascadebot/cascadebot/utils/votes/VoteButtonGroupBuilder.java

```diff
@@ -0,0 +1,188 @@
+        switch (type) {
+            case YES_NO:
+                buttonGroup.addButton(new Button.UnicodeButton(UnicodeConstants.TICK, (runner, channel, message) -> {
+                    buttonGroup.addVote(runner.getUser(), UnicodeConstants.TICK);
+                }));
+                buttonGroup.addButton(new Button.UnicodeButton(UnicodeConstants.RED_CROSS, (runner, channel, message) -> {
+                    buttonGroup.addVote(runner.getUser(), UnicodeConstants.RED_CROSS);
+                }));
+                break;
+            case NUMBERS:
+                for (int i = 1; i <= amount; i++) {
+                    char unicode = (char) (0x0030 + i); //This is setting up the first unicode character to be 003n where n is equal to i.
+                    final int num = i;
+                    buttonGroup.addButton(new Button.UnicodeButton(unicode + "\u20E3", (runner, channel, message) -> {
+                        buttonGroup.addVote(runner.getUser(), num);
+                    }));
+                }
+                break;
+            case LETTERS:
+                for (int i = 0; i < amount; i++) {
+                    char unicode = (char) (0xdde0 + (i + 6));
+                    final int num = i;
+                    buttonGroup.addButton(new Button.UnicodeButton("\uD83C" + unicode, (runner, channel, message) -> {
+                        buttonGroup.addVote(runner.getUser(), num);
+                    }));
+                }
+                break;
+            case CUSTOM:
+                for (Object object : voteButtons) {
+                    if (object instanceof Emote) {
+                        Emote emote = (Emote) object;
+                        buttonGroup.addButton(new Button.EmoteButton(emote, (runner, channel, message) -> {
+                            buttonGroup.addVote(runner.getUser(), emote.getIdLong());
+                        }));
+                    } else {
+                        String unicode = (String) object;
+                        buttonGroup.addButton(new Button.UnicodeButton(unicode, (runner, channel, message) -> {
+                            buttonGroup.addVote(runner.getUser(), unicode);
+                        }));
+                    }
+                }
+                break;
+        }
+
+        for (Button button : extraButtonList) {
+            buttonGroup.addButton(button);
+        }
+
+        new Thread(new VoteWaitRunnable(buttonGroup)).start();
+
+        return buttonGroup;
+    }
+
+    public interface IVotePeriodicRunnable {
+        void run(List<VoteResult> results, Message message);
+    }
+
+    public class VoteWaitRunnable implements Runnable {
+
+        VoteButtonGroup voteGroup;
+
+        boolean sent = false;
+
+        public VoteWaitRunnable(VoteButtonGroup buttonGroup) {
+            voteGroup = buttonGroup;
+        }
+
+        @Override
+        public void run() {
+            voteGroup.setMessageSentConsumer(() -> {
```

Comment:
What is the purpose of the `VoteWaitRunnable?`

Reply:
oh this was from the old system, before i added a consumer.

Reply:
Ah right, as far as I can see this doesn't need to be run async anymore